internal/announcements/monitors/taskmonitor.go

```diff
@@ -0,0 +1,37 @@
+package monitors
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/resonatehq/resonate/internal/announcements"
+)
+
+type taskMonitor struct {
```

Comment:
TaskMonitor must not part of the core package for announcements and monitors, but part of the test/dst -- it's application specific